app/services/suri_minter.rb

```diff
@@ -0,0 +1,56 @@
+require 'net/http'
+require 'uri'
+require 'errors'
+
+class SuriMinter < ActiveFedora::Noid::Minter::Base
+
+  protected
+
+  def next_id
+    retrieve_id
+  end
+
+  private
+
+  def valid?
+    druid.match?(/\A([a-z]{2})(\d{3})([a-z]{2})(\d{4})\z/i)
+  end
+
+  def retrieve_id
+    suri_request.basic_auth(Settings.suri.username, Settings.suri.password)
+    suri_connection.use_ssl = true
+    handle_response(suri_connection.request(suri_request))
+  rescue Net::OpenTimeout => err
+    raise Hydrox::SuriTimeoutError, "The connection to SURI has timed out."
+  end
+
+  def suri_connection
+    @suri_connection ||=  Net::HTTP.new(suri_uri.host, suri_uri.port)
+  end
+
+  def suri_uri
+    @uri ||= URI.parse(Settings.suri.service_url)
+  end
+
+  def suri_request
+    @suri_request ||= Net::HTTP::Post.new(suri_uri.request_uri)
+  end
+
+  def handle_response(response)
+    case response.code
+    when "201"
+      druid(response.body)
```

Comment:
does `response.body` return an identifier? `#druid` below seems to want an `id` arg.

Reply:
are you calling this method here just to get its side-effect of setting an ivar with the id stripped out?

Reply:
Yes, response body returns JUST and ID (no other HTML/JSON) with some whitespace (hence the split). I'm happy to rename the parameter here. This small bit was in order to pull out the `valid?` check and keep the `handle_response` short

Reply:
essentially, yes.